### ISSUE TITLE
Add a toggle to enable more precise timestamp

### ISSUE THREAD

**Issue by cecton** [sebasmagri/env_logger#87]
In my application I really need to see the exact time it takes in milliseconds between each log. This will give me an idea if something is slow.

As suggested [here](https://stackoverflow.com/questions/48850403/change-timestamp-format-used-by-env-logger), I'm asking if it would be possible to implement a toggle to have it more precise?

**Comment by KodrAus** [comment]
Thanks for the suggestion @cecton!

I've kept the timestamp formatting in `env_logger` intentionally limited so you're stuck with a format that's totally unambiguous.

Being able to increase the precision of that format sounds like a good idea though. We could do something like:

```rust
impl Builder {
    // defaults to `false`, like our current format
    fn default_format_timestamp_nanos(yes: bool);
}

impl Formatter {
    fn precise_timestamp(&self) -> PreciseTimestamp;
}

impl Timestamp {
    fn precise(&self) -> PreciseTimestamp {
        PreciseTimestamp(self.0)
    }
}

// Same as Timestamp, but uses a format like 2018-02-14T00:28:07.000000000Z
struct PreciseTimestamp(SystemTime);
```

So that when you set `default_format_timestamp_nanos(true)` you get the more precise format. The only problem with this more precise format so far would be that we don't capture it until attempting to write the timestamp, which is fairly deep into the formatting machinery. We should try and capture it sooner.

**Comment by cecton** [comment]
I would gladly try to make a PR if I could but I'm not experienced enough in Rust at the moment to make something good. I will probably give it a try at some point if nobody does before.

Btw, I just thought about something... what about a feature flag? I saw that in the earlier versions of env_logger the logs where more detailed, maybe implementing a feature flag is easier I don't know (I have no idea 😁 )

**Comment by KodrAus** [comment]
@cecton I'd be more than happy to help you through a PR if you'd like to give it a shot!

I think managing this through our builders rather than feature flags makes the most sense for `env_logger`, because we've already got a lot of machinery there around tweaking the format so it's less surprising to find the timestamp precision there too.

**Comment by cecton** [comment]
(I still plan to do it)

**Comment by ivan-ochc** [comment]
@KodrAus @cecton, hi! I also agree that we need more precise timestamp. 
I have a question regarding this issue:
is format `2018-02-14T00:28:07.000000000Z` what we need? We can use `format_rfc3339_nanos` for this purpose. If we want to format nanoseconds like% f,% .3f, etc., seems that it's required much more complicated logic or we can use `chrono` instead of` SystemTime`.

**Comment by KodrAus** [comment]
@ivan-ochc If I understand correctly you're asking if the RFC3339 format with nanos is the right one for us to use? I think it's the most appropriate format for our high-precision timestamp that means we can stick with `humantime` and avoid depending on `chrono`, which we recently removed.

**Comment by KodrAus** [comment]
This was released in [`0.5.12`](https://github.com/sebasmagri/env_logger/releases/tag/v0.5.12) :tada:
